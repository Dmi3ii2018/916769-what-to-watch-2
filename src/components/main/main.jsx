import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link, withRouter} from "react-router-dom";

import {FilmsList} from "../films-list/films-list";
import {GenresList} from "../../components/geners-list/genres-list";
import {Operation} from "../../reducer/root-reducer";

export class Main extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isFavorite: false,
    };
  }

  _promoFilmToFilm(promoId, filmList) {
    console.log(this.props);
    const film = filmList.find((it) => {
      return it.id === promoId;
    });
    return film;
  }

  render() {
    const {filmData, onHeaderClick, promoFilm, isAuthorizationRequired, renderFavoriteFilms} = this.props;
    console.log(this.props);
    const promoInFilmList = this._promoFilmToFilm(promoFilm.id, filmData);
    console.log(promoInFilmList);
    const posterBackgroundColor = {
      backgroundColor: promoFilm.background_color,
    };

    return <>
    <section className="movie-card" style={posterBackgroundColor}>
      <div className="movie-card__bg">
        <img src={`${promoFilm.background_image}`} alt="The Grand Budapest Hotel" />
      </div>

      <h1 className="visually-hidden">WTW</h1>
      <header className="page-header movie-card__head">
        <div className="logo">
          <a className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </a>
        </div>

        {isAuthorizationRequired
          ? <div className="user-block">
            <Link to="/login" className="user-block__link">Sign in</Link>
          </div>
          : <div className="user-block">
            <div className="user-block__avatar">
              <Link to="/mylist">
                <img src={`https://htmlacademy-react-2.appspot.com${this.props.avatarSrc}`} alt="User avatar" width="63" height="63" />
              </Link>
            </div>
          </div>
        }

      </header>

      <div className="movie-card__wrap">
        <div className="movie-card__info">
          <div className="movie-card__poster">
            <img src={`${promoFilm.poster_image}`} alt="The Grand Budapest Hotel poster" width="218" height="327" />
          </div>

          <div className="movie-card__desc">
            <h2 className="movie-card__title" onClick={onHeaderClick}>{promoFilm.name}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{promoFilm.genre}</span>
              <span className="movie-card__year">{promoFilm.released}</span>
            </p>

            <div className="movie-card__buttons">
              <Link to="/films/{promoFilm.id}/player" className="btn btn--play movie-card__button" type="button">
                <svg viewBox="0 0 19 19" width="19" height="19">
                  <use xlinkHref="#play-s"></use>
                </svg>
                <span>Play</span>
              </Link>

              {promoFilm.is_favorite
                ? <button onClick={() => {
                  console.log("321");
                  // checkSignInAuth(isAuthorizationRequired);
                  this.setState({isFavorite: false});
                  this.props.setFavorite(promoFilm.id, promoFilm.is_favorite);
                }} className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 18 14" width="18" height="14">
                    <use xlinkHref="#in-list"></use>
                  </svg>
                  <span>My list</span>
                </button>
                : <button onClick={() => {
                  console.log("123");
                  // checkSignInAuth(isAuthorizationRequired);
                  this.setState({isFavorite: true});
                  this.props.setFavorite(promoFilm.id, promoFilm.is_favorite);
                }} className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 19 20" width="19" height="20">
                    <use xlinkHref="#add"></use>
                  </svg>
                  <span>My list</span>
                </button>
              }

            </div>
          </div>
        </div>
      </div>
    </section>
  <div className="page-content">
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <GenresList />

      <div className="catalog__movies-list">
        <FilmsList
          films = {filmData}
        />
      </div>

      <div className="catalog__more">
        <button className="catalog__button" type="button">Show more</button>
      </div>
    </section>

    <footer className="page-footer">
      <div className="logo">
        <a className="logo__link logo__link--light">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="copyright">
        <p>Â© 2019 What to watch Ltd.</p>
      </div>
    </footer>
  </div>
  </>;
  }
}

const mapStateToProps = (state) => ({
  avatarSrc: state.authorizationReducer.avatar_url,
  isAuthorizationRequired: state.authorizationReducer.isAuthorizationRequired,
  promoFilm: state.filterReducer.promoFilm,
  filmData: state.filterReducer.filmsList,
});

const mapDispatchToProps = (dispatch) => ({
  setFavorite: (id, status) => {
    dispatch(Operation.postFavoriteFilms(id, status));
  },
});

export const MainPage = withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));

Main.propTypes = {
  onHeaderClick: PropTypes.func,
  filmData: PropTypes.array,
  avatarSrc: PropTypes.string,
  isAuthorizationRequired: PropTypes.bool,
  promoFilm: PropTypes.object.isRequired,
  setFavorite: PropTypes.func.isRequired,
};
