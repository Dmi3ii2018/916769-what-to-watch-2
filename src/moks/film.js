export const films = [
  {
    id: 0,
    genre: `Fantasy`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `fantastic-beasts-the-crimes-of-grindelwald`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 1,
    genre: `Comedian`,
    name: `Bohemian Rhapsody`,
    img: `bohemian-rhapsody`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    genre: `Crime`,
    name: `Macbeth`,
    img: `macbeth`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 3,
    genre: `Dramas`,
    name: `Aviator`,
    img: `aviator`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 4,
    genre: `Horror`,
    name: `We need to talk about Kevin`,
    img: `we-need-to-talk-about-kevin`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
