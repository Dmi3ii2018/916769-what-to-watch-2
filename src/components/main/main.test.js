import React from "react";
import {Main} from "../main/main";
import {films} from "../../moks/film";
import {shallow} from "enzyme";

it(`MainPage correctly renders after relaunch`, () => {
  const tree = shallow(<Main
    onHeaderClick = {jest.fn()}
    filmData = {films}
    avatarSrc = {`path`}
    isAuthorizationRequired = {false}
    promoFilm = {{backgroundColor: `white`}}
    setFavorite = {jest.fn()}
  />);

  expect(tree).toMatchSnapshot();
});
