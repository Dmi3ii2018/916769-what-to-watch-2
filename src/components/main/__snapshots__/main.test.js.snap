// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage correctly renders after relaunch 1`] = `ShallowWrapper {}`;
